### PR TITLE
HPCC-23231 Whiteilst thor slaves for GetUniqueInteger

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - please review

File: dali/server/daserver.cpp [hunk 1]
```diff
@@ -255,12 +255,22 @@ static bool populateWhiteListFromEnvironment(IWhiteListWriter &writer)
                 {
                     const char *masterCompName = component.queryProp("ThorMasterProcess/@computer");
                     StringBuffer ipSB;
-                    const char *ip = resolveComputer(masterCompName, component.queryProp("@netAddress"), ipSB);
+                    const char *ip = resolveComputer(masterCompName, nullptr, ipSB);
```
Comment (ghalliday) [inline dali/server/daserver.cpp]: This doesn't look like it is related.  What was the reason for this change?

**Comment by ghalliday** [comment]
@jakesmith one comment.  

Out of interest, if a slave has connected, what stops it then calling the other dali functions?

**Comment by jakesmith** [comment]
>Out of interest, if a slave has connected, what stops it then calling the other dali functions?

not a lot, but it's not a Dali client at this point, only an MP client.
So calling most Dali code would fail for that reason, as the client Coven isn't initialized, but if you could write your own implementation you could workaround it.

File: dali/server/daserver.cpp (hunk 1 above)
Comment (jakesmith) [reply dali/server/daserver.cpp]: no not really related, but spotted that that parameter isn't used - the topology of a ThorMasterProcess/ThorSlaveProcess, does not have a @netAddress, it is always looked up via the computer name.

**Comment by jakesmith** [comment]
@ghalliday - please see responses.

**Comment by ghalliday** [comment]
@jakesmith looks good.  I think it needs to target 7.4.x though.

**Comment by jakesmith** [comment]
@ghalliday - rebased. Ready to merge.

**Pull request merged**